ui/test/nuxeo-image-viewer.test.js

```diff
@@ -0,0 +1,326 @@
+      } finally {
+        expect(isViewerVisible(viewer)).to.be.false;
+        expect(viewer._el).to.be.undefined;
+      }
+    });
+
+    test('Should not render image content when a nonexistent source image is provided', async () => {
+      const viewer = await fixture(
+        html`
+          <nuxeo-image-viewer src="${base}/resources/nonexistent.png"></nuxeo-image-viewer>
+        `,
+      );
+      await flush();
+      try {
+        await waitViewerLoad(viewer, 500);
+      } finally {
+        expect(isViewerVisible(viewer)).to.be.false;
+        expect(viewer._el).to.be.undefined;
+      }
+    });
+
+    test('Should render image content when a source image is provided', async () => {
+      const viewer = await fixture(
+        html`
+          <nuxeo-image-viewer src="${base}/resources/sample.png"></nuxeo-image-viewer>
+        `,
+      );
+      await viewerLoad(viewer);
+      expect(isViewerVisible(viewer)).to.be.true;
+    });
+
+    suite('Controls Toolbar', () => {
+      test('Should not render controls toolbar when no source image is provided', async () => {
+        const viewer = await fixture(
+          html`
+            <nuxeo-image-viewer controls></nuxeo-image-viewer>
+          `,
+        );
+        await flush();
+        expect(isToolbarVisible(viewer)).to.be.false;
+      });
+
+      test('Should not render controls toolbar when a nonexistent source image is provided', async () => {
+        const viewer = await fixture(
+          html`
+            <nuxeo-image-viewer src="${base}/resources/nonexistent.png" controls></nuxeo-image-viewer>
+          `,
+        );
+        await flush();
+        expect(isToolbarVisible(viewer)).to.be.false;
+      });
+
+      test('Should render controls toolbar when a source image is provided', async () => {
+        const viewer = await fixture(
+          html`
+            <nuxeo-image-viewer src="${base}/resources/sample.png" controls></nuxeo-image-viewer>
+          `,
+        );
+        await viewerLoad(viewer);
+        expect(isToolbarVisible(viewer)).to.be.true;
+      });
+
+      test('Should display five options when toolbar is displayed', async () => {
+        const viewer = await fixture(
+          html`
+            <nuxeo-image-viewer src="${base}/resources/sample.png" controls></nuxeo-image-viewer>
+          `,
+        );
+        await viewerLoad(viewer);
+        expect(isToolbarVisible(viewer)).to.be.true;
+        const toolbar = viewer.$$('#toolbar');
+
+        const options = toolbar.children;
+        expect(options).to.have.lengthOf(5);
+      });
+
+      suite('Options', () => {
+        let viewer;
+        let toolbar;
+        setup(async () => {
+          viewer = await fixture(
+            html`
+              <nuxeo-image-viewer src="${base}/resources/sample.png" controls></nuxeo-image-viewer>
+            `,
+          );
+          await viewerLoad(viewer);
+          expect(isToolbarVisible(viewer)).to.be.true;
```

Comment:
This was already tested on the previous test. Also, I wound't expect an assert inside a setup. 🤔

Reply:
It is a good question and it can generate a good discussion.
I understand your point that it is already tested but, on the other hand, all the tests depend on that to be verified, that's why the setup.